server/controllers/card_holder_controller.py

```diff
@@ -6,8 +6,18 @@
 class CardHolderController(BaseController):
     model = CardHolder
 
-    def get_card_holders(self, search_params):
-        return self.get_all(**search_params)
+    def get_card_holders(self, first_name=None, last_name=None, card_number=None):
```

Comment:
I get that we're being explicit with these methods, but would it make more sense to consoldiate this to fewer methods that just took the queries we're searching for, instead of also having methods below, to dry this up?

Reply:
This method is not super clear to me on how it should be used - do we want to match on first_name OR last_name, or make sure only entries that match each criteria? Can we make this more explicit in the code?

Also this class & method could use a docstring to make it clearer waht we want this to return.